docs/plugins/npe2_manifest_specification.md

```diff
@@ -0,0 +1,606 @@
+(npe2-manifest-spec)=
```

Comment:
general thought on this file.  I feel like it reads somewhere in between "tutorial" and "reference" ... making it a bit less useful for both purposes.  If someone's made a plugin before and just can't remember what the manifest specification is, there should be a page where they can go to essentially see the schema, with bullet points for the fields, and probably one example.  We also certainly need all of the more tutorial-like content here, but having them all together makes it a bit more cumbersome to reference?

Reply:
I agree - I like the idea of separating tutorial/concepts/reference. I also want to add guides for the specific plugin types.

I don't want to have to address that in this PR though. It'll be too complicated. I'd like to get this in as a good starting place, then work out a set of issues so we can get the structure where we want it piece by piece.